From 4cb260e4642504467ad6601acdbfb50b77aa1ffa Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 28 Oct 2021 22:22:35 +0300
Subject: [PATCH] [SW-0] greco/gaudi2: arc: (re)use common wfe()

We have a common definition of the wfe() which resolves to real wfe
instruction on ARM and into sleep instruction on ARC.
Use the common wfe() definition for Greco and Gaudi2 ARCs.

Change-Id: Id1f5c2e9b40bf7650f55fe61d63b6ff740c28058
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index f137546..a781b1f 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -324,7 +324,7 @@
 
 	/* Put secondary CPU in sleep state until further notice */
 	while (1)
-		__asm__ __volatile__("sleep\n\t" : : );
+		wfe();
 }
 
 void preboot_pcie_wait_for_link(void)
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 69b5e49..6640138 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -294,7 +294,7 @@
 
 	/* Put secondary CPU in sleep state until further notice */
 	while (1)
-		__asm__ __volatile__("sleep\n\t" : : );
+		wfe();
 }
 
 void preboot_pcie_wait_for_link(void)
